lib/vagrant-libvirt/driver.rb

```diff
@@ -167,13 +167,19 @@ def get_ipaddress_from_qemu_agent(domain, machine_id)
         end
 
         unless addresses.nil?
-          addresses["return"].each{ |ip|
-            if domain.mac == ip["hardware-address"]
-              @logger.debug("Found mathing interface: [%s]" % ip["name"])
-              if ip.has_key?("ip-addresses")
-                ip_address = ip["ip-addresses"][0]["ip-address"]
-                @logger.debug("Return IP: [%s]" % ip_address)
-                break
+          addresses["return"].each{ |interface|
+            if domain.mac == interface["hardware-address"]
+              @logger.debug("Found mathing interface: [%s]" % interface["name"])
+              if interface.has_key?("ip-addresses")
+                interface["ip-addresses"].each{ |ip|
+                  # returning ipv6 addresses might break windows guests because
```

Comment:
I suspect that it shouldn't return the ipv6 address as an array and instead it's likely there is bug elsewhere, possibly we need to return information in a `winrm_info` with ipv6 addresses wrapped for the hostname rather than returning `[ipv6]` as the address.
vagrant will fall back to using the ssh info if no winrm info is provided for the host:
https://github.com/hashicorp/vagrant/blob/cfa8722007b1b78ea06632db499386b5593416eb/plugins/communicators/winrm/helper.rb#L31-L38

What I'm not sure about is whether this would be fixed by changing how the ssh info is formatted for ipv6, or need to return a dedicated winrm info response.

I'd stick with your approach to only return an ipv4 address for now, until it's possible to work out what the correct behaviour is around ipv6.